src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -99,41 +106,75 @@ namespace {
 NetworkInfo sNetworks[kMaxNetworks];
 } // namespace
 
-// TODO: We cannot get length of operational dataset here, should be fixed after we have byte string type.
-// Note: The codegen for OCTET_STRING will be implemented later since no other clusters are using OCTET_STRING.
 EmberAfNetworkCommissioningError OnAddThreadNetworkCommandCallbackInternal(app::Command *, EndpointId,
-                                                                           const uint8_t * operationalDataset, uint64_t breadcrumb,
+                                                                           const uint8_t * operationalDataset,
+                                                                           size_t operationalDatasetLen, uint64_t breadcrumb,
                                                                            uint32_t timeoutMs)
 {
-    // TODO: Implement AddThreadNetworkCommand after OCTET_STRING is supported.
-    return EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_NETWORK_NOT_FOUND;
+    EmberAfNetworkCommissioningError err = EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_BOUNDS_EXCEEDED;
+
+    static_assert(std::is_same<std::remove_cv_t<decltype(kMaxNetworkIDLen)>, uint8_t>::value);
+    static_assert(sizeof(kTemporaryThreadNetworkId) <= kMaxNetworkIDLen);
+    static_assert(sizeof(kTemporaryThreadNetworkId) <= sizeof(ThreadNetworkInfo::mDataset));
```

Comment:
These static_assert messages fail to compile (at least on Mac), like so:
```
error: static_assert with no message is a C++17 extension [-Werror,-Wc++17-extensions]
```
Adding messages would also allow describing what goes wrong if the assert fails.  Right now these asserts are pretty inscrutable, not least because they are not placed next to the lines of code which depend on them.  We should move them next to those lines of code.

Reply:
Done